core/insertion_marker_manager.js

```diff
@@ -279,9 +290,17 @@ Blockly.InsertionMarkerManager.prototype.createMarkerBlock_ = function(sourceBlo
         continue;  // Ignore the collapsed input.
       }
       var resultInput = result.inputList[i];
+      if (!resultInput) {
+        throw new Error(Blockly.InsertionMarkerManager.DUPLICATE_BLOCK_ERROR
+            .replace('{0}', 'an input'));
+      }
       for (var j = 0; j < sourceInput.fieldRow.length; j++) {
         var sourceField = sourceInput.fieldRow[j];
         var resultField = resultInput.fieldRow[j];
+        if (!resultField) {
+          throw new Error(Blockly.InsertionMarkerManager.DUPLICATE_BLOCK_ERROR
+              .replace('{0}', 'a field'));
```

Comment:
I would change the placeholder symbol from `'{0}'` to `'%1'` since that's a more common pattern in the code base.

Reply:
Thank you! I couldn't remember which way it's normally done in javascript.